### PR TITLE
Fixes: fix the code to return tuple so we can add doctests in it and …

### LINKED ISSUE #10189: Change Code to return tuple in dynamic_programming/subset_generation.py
### What would you like to share?

The file dynamic_programming/subset_generation.py codes printing the list, which is hard to add doctest in it. And make it cover lower 
## Changes
Change the code to return tuple so we can add doctests in it and make cover higher
### Additional information

_No response_

### PR DESCRIPTION
**Pull request by TheRealSaiTama**
…make cover higher in dynamic_programming/subset_generation.py

### Describe your change:
   1. In this version, each combination is added to a list.
   2.  The combination_util function returns a list of tuples, where each tuple represents a valid combination of r elements from the input array.
    3. It separates the process of generating and returning combinations, making it easier to work with the results, such as adding doctests.

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".
Fixes #10189

### CODE REVIEW

File: dynamic_programming/subset_generation.py [hunk 1]
```diff
@@ -1,44 +1,44 @@
-# Print all subset combinations of n element in given set of r element.
+# Return all subset combinations of n element in given set of r element.
 
 
 def combination_util(arr, n, r, index, data, i):
```
Comment (cclauss) [inline dynamic_programming/subset_generation.py:4]: Please add type hints to the function signature.

File: dynamic_programming/subset_generation.py [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
+    if res1 is None:
+        return res2
+    elif res2 is None:
+        return res1
+    else:
+        return res1, res2
 
 
-def print_combination(arr, n, r):
+def get_combinations(arr, n, r):
```
Comment (haxkd) [inline dynamic_programming/subset_generation.py:33]: provide proper name with its type and description

**Review by haxkd — CHANGES_REQUESTED** [review]

File: dynamic_programming/subset_generation.py (hunk 2 above)
Comment (TheRealSaiTama) [reply dynamic_programming/subset_generation.py:33]: i changed the code check it once

File: dynamic_programming/subset_generation.py (hunk 1 above)
Comment (TheRealSaiTama) [reply dynamic_programming/subset_generation.py:4]: i changed the code so do check it once

**Comment by cclauss** [comment]
Closing tests_are_failing PRs to prepare for Hacktoberfest 2024

**Pull request closed without merging**